program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,150 @@
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-08-01 00:00:00'
+        )
+
+        response = self.get_response(self.user, 200)
+
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['certainty'], 'CERTAIN_YES')
+
+        return response
+
+    def test_maybe_request_taken_if_no_certain(self):
+        """
+        Test that when user has no certain program intents it returns the maybe
+        """
+
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id2,
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-07-01 00:00:00'
+        )
+
+        response = self.get_response(self.user, 200)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['certainty'], 'MAYBE')
+
+    def test_user_no_intents(self):
+        """
+        Test that when user has no program intents you get 404 response
+        """
+        new_user = UserFactory()
+        self.get_response(new_user, 404)
+
+
+    def test_multiple_intents(self):
+        """
+        Test that when user has multiple program intents you get the correct responses
+        """
+        expected_intent1 = ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id1,
+            reason = 'a reason',
+            certainty = 'CERTAIN_YES',
+            effective_timestamp = '2021-07-01 00:00:00'
+        )
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id1,
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-08-01 00:00:00'
+        )
+
+        # import pudb; pu.db
+
+        response = self.get_response(self.user, 200)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['certainty'], expected_intent1.certainty)
+
+        expected_intent2 = ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id2,
+            reason = 'a reason',
+            certainty = 'CERTAIN_NO',
+            effective_timestamp = '2021-07-01 00:00:00'
+        )
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id2,
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-09-01 00:00:00'
+        )
+
+        response = self.get_response(self.user, 200)
+        self.assertEqual(len(response.data), 2)
+
+        actual_certainties = [response.data[0]['certainty'], response.data[1]['certainty']]
+
+        self.assertIn(expected_intent1.certainty, actual_certainties)
+        self.assertIn(expected_intent2.certainty, actual_certainties)
+
+    # def test_invalid_reason(self):
```

Comment:
Should the reasons people input be tested for validity?

Reply:
no, we don't want to require people to sync on that

Reply:
What do you mean by "validity" here? Do you mean whether it's one of our predefined reasons? I don't think you need to test that. We don't have any business logic yet that enforces any validation for reasons. It would make sense if your view or serializer or model enforced rules about the `reason`, but they don't.